daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/Runner.scala

```diff
@@ -340,23 +340,27 @@ private[lf] class Runner(
     timeMode: ScriptTimeMode,
 ) extends StrictLogging {
 
-  // We overwrite the definition of fromLedgerValue with an identity function.
-  // This is a type error but Speedy doesn’t care about the types and the only thing we do
-  // with the result is convert it to ledger values/record so this is safe.
   private val extendedCompiledPackages = {
-    val fromLedgerValue: PartialFunction[SDefinitionRef, SDefinition] = {
+    val replacements: PartialFunction[SDefinitionRef, SDefinition] = {
       case LfDefRef(id) if id == script.scriptIds.damlScript("fromLedgerValue") =>
+        // We overwrite the definition of fromLedgerValue with an identity function.
+        // This is a type error but Speedy doesn’t care about the types and the only thing we do
+        // with the result is convert it to ledger values/record so this is safe.
         SDefinition(SEMakeClo(Array(), 1, SELocA(0)))
+      case LfDefRef(id) if id == script.scriptIds.damlScript("validateUserIdBuiltin") =>
+        // We overwrite the definition of validateUserIdBuiltin with the corresponding speedy builtin
+        val builtin = SEBuiltin(SBValidateTextAsUserId)
```

Comment:
Very minor remark. Maybe you could declare this value as global in order to avoid its allocation multiple times.